src/os/bluestore/BlueStore.cc

```diff
@@ -5397,6 +5397,25 @@ void *BlueStore::MempoolThread::entry()
     interval_stats_trim = false;
 
     store->refresh_perf_counters();
+    uint64_t period = store->cct->_conf.get_val<uint64_t>("bluestore_fragmentation_check_period");
+    if (period != 0) {
+      auto now = mono_clock::now();
+      timespan elapsed = now - last_fragmentation_check;
+      if (elapsed > make_timespan(period)) {
+        last_fragmentation_check = now;
+        double score = 0;
+        if (store->alloc) {
```

Comment:
may be do above along with "period !=0" check:

if (period != 0 && store->alloc)
...